### DISCUSSION TITLE
Welcome to react-highlight-within-textarea Discussions!

### DISCUSSION THREAD

**Discussion by bonafideduck** [bonafideduck/react-highlight-within-textarea#5] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by ntountasakis** [comment]
Hi, thanks for making the library! In my project I'm working to add tooltips to each highlighted text section. I'm using Tippy.js and the React port of it needs a ref to elements you want to add tooltips for. I don't think this library exposes refs of the generated <mark> elements to the user, is this something you'd consider adding?

I've also noticed that dev tools complain about that the generated list of <mark> tags don't have unique keys= assigned to them. As far as I can tell this isn't a deal breaker but would be nice to fix.

**Reply by bonafideduck** [reply]
I’m not against exposing the reference. One challenge would be overlapping matches.  Currently if the string is “aba” and the matches are “ab” and “ba” this creates three spans with associated classes for each.

The unique key issue sounds like a bug.

**Reply by ntountasakis** [reply]
If I'm following along the library generates 3 spans. One for the actual text area input "aba", one for the highlighted section "ab" and another "ba". Is the concern that's it's not easy to only provide refs for the highlighted areas? Or that the user of the library would only expect a single ref here, that being the one that highlights "aba" but they'd actually be getting two?

**Reply by bonafideduck** [reply]
More like it splits it into `<mark>a</mark><mark>b</mark><mark>a</mark>`.  I've got an idea that probably will work for you that will include sending a function view as a property.  I hoped to have something today, but got into npm package upgrade pains.

Also, can you paste the unique keys= issue you are having?  I set the key for the character matched.  That could have redundant similar keys if there are multiple matches starting at the same location.

**Reply by bonafideduck** [reply]
@ntountasakis , check out the last example in https://bonafideduck.github.io/react-highlight-within-textarea/.  It still needs some cleaning up, but would that work for you.  One annoyance I haven't been able to correct is if you click on the area where the tooltip pops up, it won't move the cursor to that position in the text area.

**Reply by ntountasakis** [reply]
Hey, this looks great. Sorry about the slow reply, busy with life :)

Will try to reproduce the unique key issue, and excited to use this feature.

**Reply by bonafideduck** [reply]
@ntountasakis , Version 1.0.0 has your changes.

**Comment by coltisor** [comment]
Is it working with MUI?
